InvenTree/InvenTree/test_views.py

```diff
@@ -41,3 +42,80 @@ def test_panels(self):
         self.assertIn("<div id='detail-panels'>", content)
 
         # TODO: In future, run the javascript and ensure that the panels get created!
+
+    def test_settings_page(self):
+        """Test that the 'settings' page loads correctly"""
+
+        # Settings page loads
+        url = reverse('settings')
+
+        # Attempt without login
+        self.client.logout()
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 302)
+
+        # Login with default client
+        self.client.login(username=self.username, password=self.password)
+
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+        content = response.content.decode()
+
+        user_panels = [
+            'account',
+            'user-display',
+            'user-home',
+            'user-reports',
+        ]
+
+        staff_panels = [
+            'server',
+            'login',
+            'barcodes',
+            'currencies',
+            'parts',
+            'stock',
```

Comment:
Question: Do you see an option to get these inputs dynamically from the files?

Reply:
Not an *easy* option. 20% effort for 80% return :)